Sources/PTTOnAction.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include "DiscordESDAction.h"
+
+class PTTOnAction final : public DiscordESDAction {
+ public:
+  using DiscordESDAction::DiscordESDAction;
+  static const std::string ACTION_ID;
+  virtual std::string GetActionID() const override { return ACTION_ID; }
+  
+  virtual void KeyUp(std::shared_ptr<DiscordClient> client) override {
+    client->setIsPTT(true);
+  }
+
+  virtual int GetDesiredState(const DiscordClient::State& state) override {
+      return state.isPTT ? 0 : 1;
```

Comment:
This should be the opposite of `PTTOffAction::getDesiredState()`; I *think* this one is the wrong way round